plugin/src/main/java/io/jenkins/plugins/remotingkafka/KafkaComputerLauncher.java

```diff
@@ -103,28 +105,29 @@ private CommandTransport makeTransport(SlaveComputer computer) {
         }
         Capability cap = new Capability();
-        String producerKey = "launch", consumerKey = "launch";
-        String producerTopic = url.getHost() + "-" + url.getPort() + "-" + computer.getName()
-                + KafkaConstants.CONNECT_SUFFIX;
-        List<String> consumerTopics = Arrays.asList(computer.getName() + "-" + url.getHost() + "-" + url.getPort()
-                + KafkaConstants.CONNECT_SUFFIX);
+        String producerKey = nodeName, consumerKey = nodeName;
+        String producerTopic = url.getHost() + "-" + url.getPort() + "-" + nodeName
+                + KafkaConfigs.CONNECT_SUFFIX;
+        List<String> consumerTopics = Arrays.asList(nodeName + "-" + url.getHost() + "-" + url.getPort()
+                + KafkaConfigs.CONNECT_SUFFIX);
 
         Properties producerProps = GlobalKafkaConfiguration.get().getProducerProps();
-        if (producerProps.getProperty(KafkaConstants.BOOTSTRAP_SERVERS) == null) {
+        if (producerProps.getProperty(KafkaConfigs.BOOTSTRAP_SERVERS) == null) {
             throw new IllegalStateException("Please provide Kafka producer connection URL in global setting");
         }
-        producerProps.put(KafkaConstants.KEY_SERIALIZER, "org.apache.kafka.common.serialization.StringSerializer");
-        producerProps.put(KafkaConstants.VALUE_SERIALIZER, "org.apache.kafka.common.serialization.ByteArraySerializer");
+        producerProps.put(KafkaConfigs.KEY_SERIALIZER, "org.apache.kafka.common.serialization.StringSerializer");
+        producerProps.put(KafkaConfigs.VALUE_SERIALIZER, "org.apache.kafka.common.serialization.ByteArraySerializer");
         Producer<String, byte[]> producer = KafkaProducerClient.getInstance().getByteProducer(producerProps);
         Properties consumerProps = GlobalKafkaConfiguration.get().getConsumerProps();
-        if (consumerProps.getProperty(KafkaConstants.BOOTSTRAP_SERVERS) == null) {
+        if (consumerProps.getProperty(KafkaConfigs.BOOTSTRAP_SERVERS) == null) {
             throw new IllegalStateException("Please provide Kafka consumer connection URL in global setting");
         }
-        consumerProps.put(KafkaConstants.KEY_DESERIALIZER, "org.apache.kafka.common.serialization.StringDeserializer");
-        consumerProps.put(KafkaConstants.VALUE_DESERIALIZER, "org.apache.kafka.common.serialization.ByteArrayDeserializer");
-        KafkaConsumerPool consumerPool = KafkaConsumerPool.getInstance();
-        consumerPool.init(4, consumerProps);
-        KafkaConsumer<String, byte[]> consumer = consumerPool.getByteConsumer();
+        consumerProps.put(KafkaConfigs.GROUP_ID, "master-" + nodeName);
+        consumerProps.put(KafkaConfigs.KEY_DESERIALIZER, "org.apache.kafka.common.serialization.StringDeserializer");
+        consumerProps.put(KafkaConfigs.VALUE_DESERIALIZER, "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+        Thread.currentThread().setContextClassLoader(null);
```

Comment:
I guess without this you get errors working with Kafka classes? Might be worth reading https://jenkins.io/doc/developer/plugin-development/dependencies-and-class-loading/#context-class-loaders to see if there is anything you can do to fix it, and to make sure that you handle any failures thrown by the Kafka code correctly.